.ci/ci_cache_nemu.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+# Copyright (c) 2019 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# The main purpose of this script is to
+# optimise the nemu build from source and
+# use a cache as a prebuilt
+
+set -o errexit
+set -o nounset
+set -o pipefail
+set -o errtrace
+
+script_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+# Source to trap error line number
+source "${script_dir}/../lib/common.bash"
+
+WORKSPACE=${WORKSPACE:-$(pwd)}
+nemu_dir="/usr/local/bin/"
+
+cache_built_nemu() {
+	nemu_path="${1}"
+	real_path=$(readlink -f ${nemu_path})
+	nemu_binary=$(basename "${real_path}")
+	echo $(basename "${nemu_binary}") >  "latest"
+	cp "${real_path}" ./
+
+	sudo chown -R "${USER}:${USER}" ./
+	sha256sum "${nemu_binary}" >> "sha256sum-nemu"
+	cat sha256sum-nemu
+}
+
+mkdir -p "${WORKSPACE}/artifacts"
+pushd "${WORKSPACE}/artifacts"
+rm -f sha256sum-nemu
+cache_built_nemu "${nemu_dir}/nemu-system-x86_64"
```

Comment:
I think it would be safer to make `cache_built_nemu()` take a parameter for the directory to store the cached binary in (rather than requiring the caller to `pushd`).

Reply:
Also, I forgot to mention this hard-coded architecture.

Even if this only works on Intel, it's always best to use `arch=$(uname -m)` for all files like this.

But NEMU does in fact support [aarch64](https://github.com/intel/nemu#requirements) so that lends even more weight to using a `${arch}` variable here.